ocs_ci/ocs/ocp.py

```diff
@@ -1094,15 +1095,112 @@ def verify_cluster_operator_status(cluster_operator):
     operator_data = ocp.get(
         resource_name=f'{cluster_operator} -o json', out_yaml_format=False
     )
-    # operator_data = json.loads(oc_json)
     conditions = operator_data['status']['conditions']
     for condition in conditions:
         if condition['type'] == 'Degraded' and condition['status'] == 'True':
             log.info(f'{cluster_operator} status is Degraded')
             return False
-        if condition['type'] == 'Progressing' and condition['status'] == 'True':
+        elif condition['type'] == 'Progressing' and condition['status'] == 'True':
             log.info(f'{cluster_operator} status is Progressing')
             return False
     log.info(f'{cluster_operator} status is valid')
 
     return True
+
+
+def validate_cluster_version_status():
+    """
+    Verify OCP upgrade is completed, by checking 'oc get clusterversion'
+    status
+
+    Returns:
+        bool: False in case that one of condition flags is invalid:
+            Progressing (should be False), Failing(should be False)
+            or Available (should be True)
+
+    """
+    ocp = OCP(kind="clusterversion")
+    operator_data = ocp.get('-o json', out_yaml_format=False)
+    conditions = operator_data['items'][0].get('status').get('conditions', [])
+    for condition in conditions:
+        if condition['type'] == 'Progressing' and condition['status'] == 'True':
+            log.info('cluster version status is Progressing')
+            return False
+        elif condition['type'] == 'Failing' and condition['status'] == 'True':
+            log.info('cluster version status is Failing')
+            return False
+        elif condition['type'] == 'Available' and condition['status'] != 'True':
+            log.info('cluster status is not available')
+            return False
+
+    log.info('Cluster version validation - OK!')
+    return True
+
+
+def get_ocp_upgrade_channel():
+    """
+    Gets OCP upgrade channel
+
+    Returns:
+        str: OCP upgrade channel name
+
+    """
+    ocp = OCP(kind="clusterversion")
+    log.info("Gathering Subscription Channel information")
+    operator_version = ocp.get('-o json', out_yaml_format=False)
+    log.debug(f"cluster version: {operator_version}")
```

Comment:
why not logging this as info?

Reply:
it is json output, there version only as info in line 1153